fx/client.go

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package dosafx
+
+import (
+	"context"
+
+	"go.uber.org/fx/service"
+	"go.uber.org/yarpc"
+
+	"github.com/pkg/errors"
+	"github.com/uber-go/dosa"
+	dosarpc "github.com/uber-go/dosa/connectors/yarpc"
+)
+
+// New creates a new DOSA client that can be registered as an FX module.
+func New(d *yarpc.Dispatcher, h service.Host, entities ...dosa.DomainObject) (dosa.Client, error) {
```

Comment:
@eculver and I had a long conversation about this yesterday. We think we've come up with a better overall strategy for creating new clients that can obviate the need for users to even understand the entire entity registration process. @eculver is going to do a little work on that and see if we can't just obviate the need for most of this code.